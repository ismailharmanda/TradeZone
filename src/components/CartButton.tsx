import {TouchableOpacity} from 'react-native';
import {theme} from 'theme';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {useNavigation} from '@react-navigation/native';

export const CartButton = () => {
  const navigation = useNavigation();

  const onCartPress = () => {
    navigation.navigate('CartStack');
  };

  return (
    <TouchableOpacity onPress={onCartPress}>
      <MaterialCommunityIcons
        name="cart"
        size={26}
        color={theme.colors.primary.regular}
      />
    </TouchableOpacity>
  );
};
