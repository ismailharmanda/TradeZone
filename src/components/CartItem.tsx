import {
  Text,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
  Image,
  View,
} from 'react-native';
import {theme} from 'theme';
import {CartProduct} from 'screens/Cart/cart';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {toFixedNumber} from 'utils';

interface Props {
  product: CartProduct;
  onPlusPress: (product: CartProduct) => void;
  onMinusPress: (product: CartProduct) => void;
  onDelete: (product: CartProduct) => void;
}

const HEIGHT = Dimensions.get('window').height;

export const CartItem = ({
  onPlusPress,
  onMinusPress,
  onDelete,
  product,
}: Props) => {
  return (
    <View style={styles.container}>
      <Image
        style={styles.image}
        resizeMode="contain"
        source={{
          uri: product.image,
        }}
      />
      <View style={styles.textWrapper}>
        <Text numberOfLines={3} style={styles.textTitle}>
          {product.title}
        </Text>
        <Text numberOfLines={1} adjustsFontSizeToFit style={styles.textPrice}>
          {product.price} Ã— {product.quantity} ={' '}
          {toFixedNumber(product.price * product.quantity, 2)} TL
        </Text>
      </View>
      <View style={styles.calculatorWrapper}>
        <View style={styles.calculator}>
          <TouchableOpacity onPress={() => onMinusPress(product)}>
            <Text style={styles.calculatorButton}>-</Text>
          </TouchableOpacity>
          <Text style={styles.textQuantity}>{product.quantity}</Text>
          <TouchableOpacity onPress={() => onPlusPress(product)}>
            <Text style={styles.calculatorButton}>+</Text>
          </TouchableOpacity>
        </View>
        <TouchableOpacity onPress={() => onDelete(product)}>
          <MaterialCommunityIcons
            name="trash-can-outline"
            color={theme.colors.base.dark}
            size={26}
          />
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderRadius: theme.radius,
    overflow: 'hidden',
    backgroundColor: theme.colors.secondary.transparent,
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '100%',
    height: HEIGHT / 7,
    borderWidth: 1,
    borderColor: theme.colors.secondary.regular,
    paddingHorizontal: theme.spacing.sm,
    gap: theme.spacing.sm,
    paddingVertical: theme.spacing.sm,
  },
  calculator: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  calculatorWrapper: {
    height: '100%',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {width: '20%', height: '100%', flex: 1},
  button: {
    width: '100%',
    height: theme.height.button,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: theme.radius,
  },
  textTitle: {
    fontWeight: 'bold',
    fontSize: theme.text.size.xs.fontSize,
    lineHeight: theme.text.size.xs.lineHeight,
    color: theme.colors.base.dark,
    width: '100%',
    flexWrap: 'wrap',
  },

  textPrice: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.primary.regular,
    width: '100%',
    flexWrap: 'wrap',
  },
  textQuantity: {
    fontSize: theme.text.size.lg.fontSize,
  },
  textWrapper: {
    justifyContent: 'space-evenly',
    height: '100%',
    flex: 2,
  },
  calculatorButton: {
    fontSize: theme.text.size.lg.fontSize,
    lineHeight: theme.text.size.lg.lineHeight,
    fontWeight: 'bold',
    color: theme.colors.primary.regular,
    padding: theme.spacing.md,
  },
});
