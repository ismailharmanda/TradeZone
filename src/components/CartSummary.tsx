import {Text, StyleSheet, View} from 'react-native';
import {theme} from 'theme';
import {toFixedNumber} from 'utils';

interface Props {
  subTotal: number;
  discounts: {
    [key: string]: {
      total: number;
      summary: number;
      discountAmount: number;
      percentage: number;
    };
  };
  summary: number;
}

export const CartSummary = ({subTotal, discounts, summary}: Props) => {
  const categories: string[] = Object.keys(discounts);
  return (
    <View style={styles.container}>
      <Text style={styles.textTitle}>Sub Total: {subTotal} TL</Text>
      <View>
        {!!categories.length && (
          <Text style={styles.textDiscountTitle}>Discounts:</Text>
        )}
        {categories.map(category => {
          if (discounts[category].discountAmount === 0) {
            return null;
          }
          return (
            <View key={category}>
              <Text style={styles.textCategory}>
                <Text style={styles.textCategoryTitle}>{category}</Text>{' '}
                Category{' '}
                <Text style={styles.textCategoryPercentage}>
                  {discounts[category].percentage * 100}%
                </Text>{' '}
                - {discounts[category].discountAmount} TL
              </Text>
            </View>
          );
        })}
      </View>
      <Text style={styles.textSummary}>
        Summary:{' '}
        <Text style={styles.textSummaryPrice}>{toFixedNumber(summary, 2)}</Text>{' '}
        TL
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.secondary.transparent,
    padding: theme.spacing.sm,
    borderRadius: theme.radius,
    borderWidth: 1,
    borderColor: theme.colors.tertiary.transparent,
    width: '100%',
    gap: theme.spacing.md,
  },
  textTitle: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.base.dark,
    width: '100%',
    flexWrap: 'wrap',
  },
  textCategoryTitle: {
    color: theme.colors.tertiary.regular,
    textTransform: 'capitalize',
  },
  textCategory: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.base.dark,
  },
  textCategoryPercentage: {
    color: theme.colors.primary.regular,
  },
  textSummary: {
    fontWeight: 'bold',
    fontSize: theme.text.size.lg.fontSize,
    lineHeight: theme.text.size.lg.lineHeight,
    color: theme.colors.base.dark,
  },
  textSummaryPrice: {
    color: theme.colors.primary.regular,
  },
  textDiscountTitle: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.primary.regular,
    marginBottom: theme.spacing.xs,
  },
});
