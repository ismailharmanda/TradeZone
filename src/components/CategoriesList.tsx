import {useCallback} from 'react';
import {
  FlatList,
  ViewStyle,
  ListRenderItem,
  View,
  StyleSheet,
} from 'react-native';
import {Categories} from 'screens/Products/products';
import {theme} from 'theme';

interface Props {
  data: Categories;
  renderItem: ListRenderItem<string>;
  ListEmptyComponent: JSX.Element;
  contentContainerStyle?: ViewStyle;
  listStyle?: ViewStyle;
}

export const CategoriesList = ({
  data,
  renderItem,
  ListEmptyComponent,
  contentContainerStyle,
  listStyle,
}: Props) => {
  const keyExtractor = useCallback((item: any) => item, []);

  const ItemSeperator = useCallback(
    () => <View style={styles.itemSeperator} />,
    [],
  );

  return (
    <View testID="categoriesList" style={styles.container}>
      <FlatList
        testID="flatList"
        showsHorizontalScrollIndicator={false}
        ItemSeparatorComponent={ItemSeperator}
        horizontal
        style={[listStyle]}
        contentContainerStyle={contentContainerStyle}
        keyExtractor={keyExtractor}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: theme.spacing.xl,
  },
  itemSeperator: {
    width: theme.spacing.md,
  },
});
