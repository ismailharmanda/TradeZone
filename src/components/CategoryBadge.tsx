import {Text, StyleSheet, TouchableWithoutFeedback, View} from 'react-native';
import {theme} from 'theme';

interface Props {
  onPress: (item: string) => void;
  activeCategory: string;
  item: string;
}

export const CategoryBadge = ({onPress, activeCategory, item}: Props) => {
  const isBadgeActive = activeCategory === item;
  return (
    <TouchableWithoutFeedback onPress={() => onPress(item)}>
      <View
        style={[
          styles.container,
          isBadgeActive && {
            backgroundColor: theme.colors.primary.regular,
          },
        ]}>
        <Text
          style={[styles.text, activeCategory === item && styles.textActive]}>
          {item}
        </Text>
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    borderRadius: theme.radius,
    borderWidth: 0.2,
    borderColor: theme.colors.primary.regular,
  },
  text: {
    padding: theme.spacing.sm,
    textTransform: 'uppercase',

    color: theme.colors.primary.regular,
  },
  textActive: {
    color: theme.colors.base.white,
    textDecorationLine: 'none',
  },
});
