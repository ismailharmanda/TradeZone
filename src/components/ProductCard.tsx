import {
  Text,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
  Image,
} from 'react-native';
import {theme} from 'theme';
import {Product} from 'screens/Products/products';

interface Props {
  onPress?: () => void;
  product: Product;
}

const HEIGHT = Dimensions.get('window').height;

export const ProductCard = ({onPress, product}: Props) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Image
        style={styles.image}
        resizeMode="contain"
        source={{
          uri: product.image,
        }}
      />
      <Text numberOfLines={1} style={styles.text}>
        {product.title}
      </Text>
      <Text style={styles.text}>{product.price} TL</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: theme.radius,
    overflow: 'hidden',
    backgroundColor: theme.colors.secondary.transparent,
    justifyContent: 'space-evenly',
    alignItems: 'center',
    width: HEIGHT / 10,
    height: HEIGHT / 7,
    borderWidth: 1,
    borderColor: theme.colors.secondary.regular,
  },
  image: {width: HEIGHT / 10, height: HEIGHT / 10},
  button: {
    width: '100%',
    height: theme.height.button,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: theme.radius,
  },
  primary: {
    backgroundColor: theme.colors.primary.regular,
  },
  'primary-outline': {
    backgroundColor: theme.colors.base.white,
    borderWidth: 1,
    borderColor: theme.colors.primary.regular,
  },
  disabled: {
    backgroundColor: theme.colors.primary.transparent,
  },
  text: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.base.dark,
    width: '100%',
    flexWrap: 'wrap',
  },
});
