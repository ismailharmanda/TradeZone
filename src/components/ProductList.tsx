import {useCallback} from 'react';
import {
  FlatList,
  ViewStyle,
  Text,
  ListRenderItem,
  View,
  StyleSheet,
} from 'react-native';
import {Product} from 'screens/Products/products';
import {theme} from 'theme';

interface Props {
  category: string;
  data: Product[];
  renderItem: ListRenderItem<Product>;
  ListEmptyComponent: JSX.Element;
  contentContainerStyle?: ViewStyle;
  listStyle?: ViewStyle;
  activeCategory?: string;
}

export const ProductsList = ({
  category,
  data,
  renderItem,
  ListEmptyComponent,
  contentContainerStyle,
  listStyle,
  activeCategory,
}: Props) => {
  const keyExtractor = useCallback((item: any) => item.id.toString(), []);

  const isCategoryActive = activeCategory === category;

  const ItemSeperator = useCallback(
    () => <View style={styles.itemSeperator} />,
    [],
  );

  return (
    <View
      style={[
        styles.container,
        !!activeCategory && activeCategory !== category && styles.hide,
      ]}>
      <Text style={styles.text}>{category}</Text>
      <FlatList
        ItemSeparatorComponent={ItemSeperator}
        horizontal={!isCategoryActive}
        style={listStyle}
        contentContainerStyle={[
          contentContainerStyle,
          isCategoryActive && styles.activeCategoryContentContainer,
        ]}
        keyExtractor={keyExtractor}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {flexShrink: 1},
  activeCategoryContentContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    gap: theme.spacing.md,
    justifyContent: 'space-between',
  },
  text: {
    fontWeight: 'bold',
    fontSize: theme.text.size.md.fontSize,
    lineHeight: theme.text.size.md.lineHeight,
    color: theme.colors.tertiary.regular,
    textTransform: 'uppercase',
    marginBottom: theme.spacing.md,
  },
  itemSeperator: {
    width: theme.spacing.md,
  },
  hide: {
    display: 'none',
  },
});
