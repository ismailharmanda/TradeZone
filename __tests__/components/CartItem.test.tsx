import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';
import {CartItem} from 'components/CartItem';

jest.mock(
  'react-native-vector-icons/MaterialCommunityIcons',
  () => 'MaterialCommunityIcons',
);

describe('CartItem', () => {
  const mockProduct = {
    id: 1,
    title: 'Product 1',
    price: 10,
    quantity: 2,
    image: 'https://example.com/image.png',
    category: 'Category 1',
    description: 'Description 1',
    rating: {
      rate: 4.5,
      count: 10,
    },
  };

  const mockProps = {
    product: mockProduct,
    onPlusPress: jest.fn(),
    onMinusPress: jest.fn(),
    onDelete: jest.fn(),
    onPress: jest.fn(),
  };

  it('renders correctly', () => {
    const {toJSON} = render(<CartItem {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when TouchableOpacity is pressed', () => {
    const {getByTestId} = render(<CartItem {...mockProps} />);
    const touchableOpacity = getByTestId('touchable');
    fireEvent.press(touchableOpacity);
    expect(mockProps.onPress).toHaveBeenCalledWith(mockProduct);
  });

  it('calls onMinusPress when "-" button is pressed', () => {
    const {getByText} = render(<CartItem {...mockProps} />);
    const minusButton = getByText('-');
    fireEvent.press(minusButton);
    expect(mockProps.onMinusPress).toHaveBeenCalledWith(mockProduct);
  });

  it('calls onPlusPress when "+" button is pressed', () => {
    const {getByText} = render(<CartItem {...mockProps} />);
    const plusButton = getByText('+');
    fireEvent.press(plusButton);
    expect(mockProps.onPlusPress).toHaveBeenCalledWith(mockProduct);
  });

  it('calls onDelete when trash icon is pressed', () => {
    const {getByTestId} = render(<CartItem {...mockProps} />);
    const trashIcon = getByTestId('trash-icon');
    fireEvent.press(trashIcon);
    expect(mockProps.onDelete).toHaveBeenCalledWith(mockProduct);
  });

  it('renders the correct product title', () => {
    const {getByText} = render(<CartItem {...mockProps} />);
    const productTitle = getByText(mockProduct.title);
    expect(productTitle).toBeDefined();
  });

  it('renders the correct product price', () => {
    const {getByText} = render(<CartItem {...mockProps} />);
    const productPrice = getByText(
      `${mockProduct.price} Ã— ${mockProduct.quantity} = ${
        mockProduct.price * mockProduct.quantity
      } TL`,
    );
    expect(productPrice).toBeDefined();
  });

  it('calls the correct functions with the correct arguments when buttons are pressed', () => {
    const {getByText, getByTestId} = render(<CartItem {...mockProps} />);
    const minusButton = getByText('-');
    const plusButton = getByText('+');
    const trashIcon = getByTestId('trash-icon');

    fireEvent.press(minusButton);
    fireEvent.press(plusButton);
    fireEvent.press(trashIcon);

    expect(mockProps.onMinusPress).toHaveBeenCalledWith(mockProduct);
    expect(mockProps.onPlusPress).toHaveBeenCalledWith(mockProduct);
    expect(mockProps.onDelete).toHaveBeenCalledWith(mockProduct);
  });

  it('renders the product image correctly', () => {
    const {getByTestId} = render(<CartItem {...mockProps} />);
    const productImage = getByTestId('product-image');
    expect(productImage.props.source.uri).toBe(mockProduct.image);
  });

  it('renders the correct quantity when minus button is pressed', async () => {
    const {getByText} = render(<CartItem {...mockProps} />);
    const minusButton = getByText('-');
    await fireEvent.press(minusButton);
    const productQuantity = getByText(`${mockProduct.quantity}`);
    expect(productQuantity).toBeDefined();
  });
});
