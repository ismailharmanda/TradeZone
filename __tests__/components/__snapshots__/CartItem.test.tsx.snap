// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(240,237,246,0.6)",
      "borderColor": "rgb(240,237,246)",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "gap": 8,
      "height": 190.57142857142858,
      "justifyContent": "flex-start",
      "opacity": 1,
      "overflow": "hidden",
      "paddingHorizontal": 8,
      "paddingVertical": 8,
      "width": "100%",
    }
  }
  testID="touchable"
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "https://example.com/image.png",
      }
    }
    style={
      {
        "flex": 1,
        "height": "100%",
        "width": "20%",
      }
    }
    testID="product-image"
  />
  <View
    style={
      {
        "flex": 2,
        "height": "100%",
        "justifyContent": "space-evenly",
      }
    }
  >
    <Text
      numberOfLines={3}
      style={
        {
          "color": "#5A5A5A",
          "flexWrap": "wrap",
          "fontSize": 12,
          "fontWeight": "bold",
          "lineHeight": 14.4,
          "width": "100%",
        }
      }
    >
      Product 1
    </Text>
    <Text
      adjustsFontSizeToFit={true}
      numberOfLines={1}
      style={
        {
          "color": "rgb(255,119,5)",
          "flexWrap": "wrap",
          "fontSize": 14,
          "fontWeight": "bold",
          "lineHeight": 16.8,
          "width": "100%",
        }
      }
    >
      10
       Ã— 
      2
       =
       
      20
       TL
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(255,119,5)",
              "fontSize": 20,
              "fontWeight": "bold",
              "lineHeight": 21,
              "padding": 16,
            }
          }
        >
          -
        </Text>
      </View>
      <Text
        style={
          {
            "fontSize": 20,
          }
        }
      >
        2
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(255,119,5)",
              "fontSize": 20,
              "fontWeight": "bold",
              "lineHeight": 21,
              "padding": 16,
            }
          }
        >
          +
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <MaterialCommunityIcons
        color="#5A5A5A"
        name="trash-can-outline"
        size={26}
        testID="trash-icon"
      />
    </View>
  </View>
</View>
`;
